Maintenance: CSS flex guide: updates to guide 4 (#34183)
mdn/content: files/en-us/web/css/css_flexible_box_layout/ordering_flex_items/index.md

- [The Responsive Order Conflict for Keyboard Focus](https://alastairc.uk/2017/06/the-responsive-order-conflict/) via alastairc.uk (2017)

## Use cases for `order`

There are some use cases for which the fact that the logical and therefore reading order of flex items is separate from the visual order, is helpful. Used carefully, the `order` property can allow for some useful common patterns to be easily implemented.

You might have a design, perhaps a card that will display a news item. The heading of the news item is the key thing to highlight and would be the element that a user might jump to if they were tabbing between headings to find the content they wanted to read. The card also has a date; the finished design we want to create is something like this.

![A design component with a date, then heading and then content.](order-card.png)

Visually, the date appears above the heading, in the source. However, if the card was read out by a screen reader I would prefer that the title was announced first and then the publication date. We can accomplish this with the `order` property.

> The card is our flex container, with `flex-direction` set to `colum`n. The date is given an `order` of `-1`. This places it up above the heading.

Reviewer: Small typo for column, reword only if you prefer it:

```suggestion
The card is our flex container, with `flex-direction` set to `column`. We give the date an `order` of `-1`, placing it above the heading.
```

{{EmbedGHLiveSample("css-examples/flexbox/order/usecase-order.html", '100%', 730)}}

> These small tweaks are the sort of cases where the `order` property makes sense. Keep the logical order as the reading and tab order of the document, and maintain that in the most accessible and structured fashion. Then use `order` for purely visual design tweaks. Don't reorder items that receive keyboard focus. Ensure you always test your content using only a keyboard, rather than a mouse or a touchscreen; it will be apparent if your development choices make getting around the content difficult.

Reviewer: Small suggestion for the closing sentence:

```suggestion
These small tweaks are the sort of cases where the `order` property makes sense. Keep the logical order the same as the reading and tab order of the document, and maintain that in the most accessible and structured fashion. Then use `order` for purely visual design tweaks. Don't reorder items that receive keyboard focus. Ensure you always test your content using only a keyboard rather than a mouse or a touchscreen; this will reveal if your development choices make it more complex to navigate the content.
```

Reviewer: Not your edit, but I'm not really clear on the meaning of "Keep the logical order the same as the reading and tab order of the document, and maintain that in the most accessible and structured fashion."

Especially the second half. I'm not sure how to improve it, though 🤔
